ARROW-16264: [C++][CI] Valgrind timeout in arrow-compute-hash-join-node-test

westonpace opened the pull request.

westonpace commented:
@github-actions crossbow submit test-conda-cpp-valgrind

kou commented on cpp/src/arrow/compute/exec/bloom_filter_test.cc:
```diff
@@ -364,8 +364,13 @@ void TestBloomLarge(BloomFilterBuildStrategy strategy, int64_t num_build, bool u
 
 TEST(BloomFilter, Basic) {
   std::vector<int64_t> num_build;
+#if defined(THREAD_SANITIZER) or defined(ARROW_VALGRIND)
```
Could you use `||` instead of `or` in `#if`?

westonpace replied:
Yes.  I made this change.  Thank you.

westonpace commented:
@github-actions crossbow submit test-conda-cpp-valgrind

westonpace commented:
I rebased this on top of the tpc-h generator cleanup since those valgrind failures were blocking the test of this feature.  I converted this to draft as we should merge the other fix first and then I'll convert it back to ready-for-review.

westonpace commented:
CC @michalursa Valgrind started timing out on the hash-join-node-test.  The root cause for the newly emerged failure may have been the addition of the bloom filter tests or it may have been that I accidentally made the hash join node tests slow.  Either way, I think it's probably fine to shorten the size of inputs we use when testing via valgrind.

westonpace commented:
@github-actions crossbow submit test-conda-cpp-valgrind

westonpace commented:
I'm going to move forward with this to unblock valgrind.  If we decide later we want to extend the test set on valgrind we can fix that in a follow-up.
